editor/assets/default_renderpipeline/builtin-pipeline.ts

```diff
@@ -178,9 +188,17 @@ function setupCameraConfigs(
 
     cameraConfigs.useFullPipeline = (camera.visibility & (Layers.Enum.DEFAULT)) !== 0;
 
-    cameraConfigs.enableShadowMap = camera.scene
-        ? camera.scene.mainLight !== null && camera.scene.mainLight.shadowEnabled
-        : false;
+    cameraConfigs.enableMainLightShadowMap = pipelineConfigs.shadowEnabled
+        && !pipelineConfigs.usePlanarShadow
+        && camera.scene !== null
+        && camera.scene.mainLight !== null
```

Comment:
Would `camera.scene` or `camera.scene.mainLight` be `undefined`?
I found that we just check whether they're equal to `null` but did not consider `undefined`.
Maybe just using `&& camera.scene && camera.scene.mainLight && camera.scene.mainLight.shadowEnabled` which also could reduce some bytes.

Reply:
The type is `T | null`, so I just checked `null`. I will make the change.

Reply:
Changed.